Core/RoverControllerPresets/DriveControllers/DirectDriveController.cs

```diff
@@ -14,21 +17,45 @@ public RoverControl CalculateMoveVector()
 			Convert.ToSingle(LocalSettings.Singleton.Joystick.Deadzone)
 		);
 
-		Vector2 vec = new(
-			Input.GetAxis("rover_move_backward", "rover_move_forward"),
-			Input.GetAxis("rover_move_right", "rover_move_left")
-		);
-
-		if (LocalSettings.Singleton.SpeedLimiter.Enabled)
-			vec.X *= LocalSettings.Singleton.SpeedLimiter.MaxSpeed;
-
-		vec.X = Mathf.IsEqualApprox(vec.X, 0f, joyDeadZone) ? 0 : vec.X;
+		Vector3 vec;
+
+		switch (Mode)
+		{
+			case KinematicMode.Ackermann:
+				vec = new(
+					Input.GetAxis("rover_move_backward", "rover_move_forward"),
+					Input.GetAxis("rover_move_right", "rover_move_left"),
+					0);
+				break;
+			case KinematicMode.Crab:
+				vec = new(
+					Input.GetAxis("rover_move_backward", "rover_move_forward"),
+					Input.GetAxis("rover_move_right", "rover_move_left"),
+					Input.GetAxis("rover_move_down", "rover_move_up")
+				);
+				break;
+			case KinematicMode.Spinner:
+				vec = new(
+					Input.GetAxis("rover_move_backward", "rover_move_forward"),
+					0f,
+					0f
+				);
+				break;
+			case KinematicMode.EBrake:
+				vec = new(0, 0, 0); //todo wheels in X pattern?
+				break;
+			default:
+				vec = new(0, 0, 0);
+				break;
+		}
+
+		vec.X *= SpeedModifier;
+
+		vec.X = Mathf.IsEqualApprox(vec.X, 0f, joyDeadZone * SpeedModifier) ? 0 : vec.X;
 		vec.Y = Mathf.IsEqualApprox(vec.Y, 0f, joyDeadZone) ? 0 : vec.Y;
+		vec.Z = Mathf.IsEqualApprox(vec.Z, 0f, joyDeadZone) ? 0 : vec.Z;
 
-		if (Input.IsActionPressed("camera_zoom_mod"))
-			vec.X /= 4f;
-
-		return RoverControlVec2Extension.FromVector2(vec);
+		return RoverControlVec3Extension.ToRoverControl(vec);
```

Comment:
Prefer using extension methods from extended classes.
```suggestion
		return vec.ToRoverControl();
```
see:
https://github.com/wisniax/RoverControlApp/blob/3b3e3cfab920852433be6b94e847b16dd4c70398/Core/RoverControllerPresets/MqttClasses.RoverControlVec3Extension.cs#L12-L13